devito/operator.py

```diff
@@ -485,12 +428,82 @@ def _profile_output(self, arguments):
 
     def _profile_sections(self, nodes, parameters):
         """Introduce C-level profiling nodes within the Iteration/Expression tree."""
-        nodes, profiler = create_profile(nodes)
+        nodes, profiler = create_profile('timers', nodes)
         self._globals.append(profiler.cdef)
-        parameters.append(Object(profiler.varname, profiler.dtype, profiler.setup()))
+        parameters.append(Object(profiler.name, profiler.dtype, profiler.new))
         return nodes, profiler
 
 
+# Functions collecting information from a bag of expressions
```

Comment:
Note for reviewers: this whole block of code is just stuff being moved, nothing new added